hs/src/Reach/Simulator/Core.hs

```diff
@@ -154,6 +154,9 @@ type Token = Integer
 
 data Ledger = Ledger
   { nw_ledger :: M.Map Account (M.Map Token Balance)
+  -- QUESTION: do you still want nw_next_acc here as
+  -- opposed to in `Global`?
+  -- I would rather not nest data unless for a good reason :)
```

Comment:
I don't think it really matters. I can imagine `Ledger` going completely away and going into `Global`. I can also imagine some sense of abstraction purity where you have everything "about the ledger" together.